### ISSUE TITLE
Custom Fusion Builder Element

### ISSUE THREAD

**Issue by jvwekken** [Theme-Fusion/Fusion-Builder-Sample-Add-On#36]
Hello, I am trying to build a custom fusion builder element according to the documentation here https://theme-fusion.com/documentation/fusion-builder/configure/add-new-element-fusion-builder/ however I cannot get this to work and was hoping I could get some assistance. I contacted support at theme fusion and was instructed to post my question here.

The documentation says to first create a shortcode, which I have done and confirmed that it is working. I have then used the sample code as the basis to create a fusion builder element from that shortcode, however, nothing shows up in fusion builder so clearly I am doing something wrong. The code I have so far is listed below.

```php
// OFFER SHORTCODE

function offer( $atts ) {
	$a = shortcode_atts( array(
		'offer-logo' => '',
		'offer-description' => '',
		'offer-image' => '',
		'offer-link' => '',
	), $atts );

	return '<div class="offer"><div class="offer-details"><div class="offer-logo"><img src="' . esc_attr($a['offer-logo']) . '" /></div><div class="offer-description"><h3>' . esc_attr($a['offer-description']) . '</h3></div><a href="' . esc_attr($a['offer-link']) . '" class="offer-link">GET DEAL</a></div><div class="offer-image" style="background-image:url(' . esc_attr($a['offer-image']) . ');"></div></div>';
}
add_shortcode("fusion_offer", "offer");

// OFFER FUSION ELEMENT

function fusion_element_offer() {

fusion_builder_map( 
    array(
        'name'            => esc_attr__( 'Offer', 'fusion-builder' ),
        'shortcode'       => 'fusion_offer',
        'icon'            => 'fusiona-font',
        'allow_generator' => true,
        'params'          => array(
            array(
                'type'        => 'textfield',
                'heading'     => esc_attr__( 'Offer Description', 'fusion-builder' ),
                'description' => esc_attr__( 'Enter offer description', 'fusion-builder' ),
                'param_name'  => 'offer-description',
                'value'       => esc_attr__( '', 'fusion-builder' ),
            ),
			array(
                'type'        => 'textfield',
                'heading'     => esc_attr__( 'Offer Link', 'fusion-builder' ),
                'description' => esc_attr__( 'Enter offer link', 'fusion-builder' ),
                'param_name'  => 'offer-link',
                'value'       => esc_attr__( '', 'fusion-builder' ),
            ),
			array(
					'heading'     => __( 'Offer Logo', 'fusion-builder' ),
					'description' => __( 'Upload the image you would like to use for the logo', 'fusion-builder' ),
					'value'       => '',
					'type'        => 'upload',
					'param_name'  => 'offer-logo',
				),
				array(
					'heading'     => __( 'Offer Image', 'fusion-builder' ),
					'description' => __( 'Upload the image you would like to use for the offer image', 'fusion-builder' ),
					'value'       => '',
					'type'        => 'upload',
					'param_name'  => 'offer-image',
				),
        ),
    ) 
);
}
add_action( 'fusion_builder_before_init', 'fusion_element_offer' );
```

**Comment by mikka23** [comment]
Hi @jvwekken 

The code there looks correct.  The issue here will likely be the element activation.  If you go to Fusion Builder > Settings in the WordPress admin area, you should see a list of element which you can enable/disable.  Most likely the new offer element needs checked here.

You can also add auto activation code if desired, eg:
```php
fusion_builder_auto_activate_element( 'fusion_offer' );
```

**Comment by budley** [comment]
Just wanted to say thanks for posting your example code - I got the auto-activation right but had an errant setting which made it not appear. If anybody comes across this post and the activation doesn't show your element in the element picker then check if you have a parameter of:

'hide_from_builder' => false,

set in fusion_builder_map(  parameters.
For me that STOPPED it being shown.

if I included

'hide_from_builder' => true,

It ALSO hid the element.

Basically it appears that flag isn't true/false, but simply IF you include it then that element is hidden, no matter what the value. Incorrect documentation from Avada???

**Comment by mikka23** [comment]
Hi,

Yes that is correct, if that flag is present it will be treated that way.  The actual value doesn't change that right now.

**Comment by budley** [comment]
Thanks for confirming that mikka23 (hope you are well). Ooo I'm so excited that I can now create custom elements for my customers to make it easier for them to add lovely content easily. Wow Avada is powerful :-)

**Comment by dhruv-trionn** [comment]
Please give me an example of a repeater field and in it I want two upload fields and two text fields.
